lib/vmpooler/api/v1.rb

```diff
@@ -375,8 +390,16 @@ def atomically_allocate_vms(payload)
       content_type :json
       result = { 'ok' => false }
 
-      if jdata and !jdata.empty?
-        result = atomically_allocate_vms(jdata)
+      if jdata
+        empty = jdata.delete('empty')
+        invalid = jdata.delete('invalid')
+        statsd.increment(statsd_prefix + '.checkout.empty', empty) if empty
+        statsd.increment(statsd_prefix + '.checkout.invalid', invalid) if invalid
+        unless jdata.empty?
```

Comment:
Maybe the name I'm proposing isn't great, but I could see some encapsulation here:

``` ruby
  update_allocation_statistics(jdata)

  # ...

  def update_allocation_statistics(vm_requests)
      empty = jdata.delete('empty')
      invalid = jdata.delete('invalid')
      statsd.increment(statsd_prefix + '.checkout.empty', empty) if empty
      statsd.increment(statsd_prefix + '.checkout.invalid', invalid) if invalid
  end
```

Not blocking, could be done in a refactoring PR.

Similarly, @mckern is right that there's a (not a blocker, maybe refactoring PR) an opportunity for wider reuse in the way we access statsd.  Vague recollections, but in $past_job we had an interface that was more like:

```
  stats.increment("checkout.empty")
  stats.measure("checkout.frobnitz", 32.5)
  stats.time("checkout.duration") do
    some_time_consuming_operation(args)
  end
```

Where `stats` is a singleton-y instance of a Stats class, could be `stats` (some method/attr_reader), `$stats`, `puppetlabs.stats`, etc.

One could either specify prefix at the per-instantiation (once for All The Things) of this original thing, in which case it's prefix-relative; or, if the model is more varied, the keys are full paths. Here we're kind of doing "full paths", with prefix merged in at every use. Similarly for graphite/statsd/brubeck destination setup.

Encapsulating all the stats setup, etc., down to a class that encapsulates all the things and just proffers that 3-way interface for recording the things we need worked well for general reuse.